packages/scene-graph-cocos-animation/client/src/SceneGraphAnimation.ts

```diff
@@ -139,40 +140,93 @@ export default class SceneGraphAnimation implements sgmed.SceneExporterPlugin, s
     return animationComponents;
   }
 
-
-  private extendNodeWithAnimationComponent(
+  private extendNodesWithAnimationComponent(
     animationFilesMap: Map<MetaUuid, AnimationFiles>,
-    nodeRef: Node,
-    componentRef: Fire.AnimationComponentReference
+    animationComponents: Map<NodeId, Fire.AnimationComponentReference>,
+    graph: SchemaJson
   ): void {
-    nodeRef.animations = [];
+    const nodeNameMap = new Map<string, Node>();
+    for (let i = 0; i < graph.scene.length; i++) {
+      const node = graph.scene[i];
+      nodeNameMap.set(node.name, node);
+    }
 
-    for (let i = 0; i < componentRef._clips.length; i++) {
-      const animationFiles = animationFilesMap.get(componentRef._clips[i].__uuid__);
-      if (!animationFiles) {
-        continue;
-      }
-      const content = fs.readFileSync(animationFiles.anim).toString();
-      const clipJson = JSON.parse(content) as Fire.Animation;
-
-      const animation: Types.Animation = {
-        sample: clipJson.sample,
-        speed: clipJson.speed,
-        url: animationFiles.anim,
-        curves: {}
-      };
+    for (let i = 0; i < graph.scene.length; i++) {
+      const nodeRef = graph.scene[i];
+      const componentRef = animationComponents.get(nodeRef.id);
 
-      const properties = Object.keys(clipJson.curveData.props);
-      for (let j = 0; j < properties.length; j++) {
-        const property = properties[j];
-        animation.curves[property] = this.createAnimationFrames(clipJson.curveData.props[property]);
-      }
+      if (!componentRef) continue;
+
+      for (let j = 0; j < componentRef._clips.length; j++) {
+        const animationFiles = animationFilesMap.get(componentRef._clips[j].__uuid__);
+        if (!animationFiles) {
+          continue;
+        }
+        const content  = fs.readFileSync(animationFiles.anim).toString();
+        const clipJson = JSON.parse(content) as Fire.Animation;
+
+        if (clipJson.curveData.props) {
+          const animation: Types.Animation = {
+            duration: clipJson._duration,
+            sample: clipJson.sample,
+            speed: clipJson.speed,
+            url: animationFiles.anim,
+            curves: {}
+          };
+
+          const props = clipJson.curveData.props;
+          const propertyNames = Object.keys(props);
+          for (let k = 0; k < propertyNames.length; k++) {
+            const property = propertyNames[k];
+            animation.curves[property] = this.createAnimationFrames(props[property], property);
+          }
+
+          if (!nodeRef.animations) {
+            nodeRef.animations = [];
+          }
+          nodeRef.animations.push(animation);
+        }
+
+        if (clipJson.curveData.paths) {
+          const paths = clipJson.curveData.paths;
+          const nodePaths = Object.keys(paths);
+
+          for (let k = 0; k < nodePaths.length; k++) {
+            const nodePath = nodePaths[k];
+            const nodeName = nodePath.split('/').pop();
```

Comment:
I understand that those lines are not thoughtful.
This should be refactored until next release.